File: packages/sdk-java/src/test/java/io/suprawall/SupraWallTest.java
package io.suprawall;

import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

public class SupraWallTest {
    @Test
    public void testInitFailsWithoutKey() {
        SupraWall.setApiKey(null);
        assertThrows(IllegalArgumentException.class, () -> {
            SupraWall.evaluate("test", "tool", null);
        });
    }

    @Test
    public void testTestModeBypass() throws Exception {
        SupraWall.setApiKey("sw_test_123");
        SupraWall.EvaluateResponse res = SupraWall.evaluate("test", "tool", null);
        assertEquals("ALLOW", res.decision);
    }
}

File: packages/sdk-php/tests/SupraWallTest.php
<?php

use PHPUnit\Framework\TestCase;
use SupraWall\SupraWall;

class SupraWallTest extends TestCase {
    public function testEvaluateFailsWithoutApiKey() {
        $this->expectException(Exception::class);
        $this->expectExceptionMessage("SUPRAWALL_API_KEY is required");
        SupraWall::setApiKey("");
        SupraWall::evaluate("test-agent", "test-tool");
    }

    public function testTestModeApiKey() {
        SupraWall::setApiKey("ag_test_123");
        $result = SupraWall::evaluate("test-agent", "test-tool");
        $this->assertEquals("ALLOW", $result['decision']);
    }
}
